### PR TITLE
uses a proper programming language

### PR DESCRIPTION
**Pull request by favadi**
As we all know real programmers don't use golang, I reimplement this program in a real programming language instead.

### CODE REVIEW

File: gosearch.awk [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	cat_filter=ARGV[3]
+    }
+}
+
+/^## / {
+    sub(/^## /, "")
+    cat=$0
+}
+
+/^\* \[.+\]\(.+\) - (.+)$/{
```
Comment (hvnsweeting) [inline gosearch.awk:24]: no, this will match whitespaces, we don't want to see

```
[xxx](cat) -         AWKWKWKWKWKKW
```

File: gosearch.awk [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    FS="^\* \[|\]\(|\) - "
+
+    if (length(ARGV[2]) == 0) {
+	print "missing search term"
+	exit 1
+    }
+    search_term=ARGV[2]
+
+    if (length(ARGV[3]) != 0) {
+	cat_filter=ARGV[3]
```
Comment (hvnsweeting) [inline gosearch.awk:15]: -c is not category filter, it's list packages by category

**Review by hvnsweeting — COMMENTED** [review]
Awesome MR, but the real programming language is Assembly.
Suggest you to create new repo for Awesome AWK if that ever exists :1st_place_medal:

**Comment by favadi** [comment]
> Awesome MR, but the real programming language is Assembly.

I will create a PR in assembly soon.

**Pull request closed without merging**